### DISCUSSION TITLE
Change="Delete" order query

### DISCUSSION THREAD

**Discussion by p-m-j** [KevinJump/uSync#589] [General]
Imagine a site with a number of block lists all with identical config each used by different document types.

+ Doctype A
  + Blockist A
+ Doctype B
  + Blocklist B
  
What should happen if a change is made to swtich all the document types to use a single block list and the other datatypes are deleted.

+ Doctype A
  + Blockist A
+ Doctype B
  + Blockist A
  + ~Blocklist B~

When trying this out what appeared to happen on deployment is that the properties lost their value but ended up with correct configuration.

Is this a known limitation or a weird unexpected issue?

I'm guessing that what happened is that the deletes of the datatypes occurred before the update of the document types resulting in accidental data loss but it's just a guess.

If this is what happens would a two pass approach work where all deletes are processed on the second pass allowing for updates to take place first?

**Comment by KevinJump** [comment]
Hi, 

The deletes should already happen at the end, but the switching of the type, will probably trigger the deletes inside Umbraco 😞 

when you change a doctype that triggers all the internal events in umbraco and if a property is change umbraco updates all the content across the board.

I am not sure there is much we can do about it, but i will look.

**Reply by p-m-j** [reply]
The next part of this story is that I did a point in time restore of the database, reverted the commit which deleted the datatypes leaving only the commit which switched the document types to use a different datatype for their block list prop.

deploying this did not result in any data loss but does leave a bunch of unused datatypes to be removed by a later release.

So changing the datatype of the property alone did not result in any data loss.

**Reply by p-m-j** [reply]
@KevinJump was looking through release notes and spotted v10.6.2 has a fix for https://github.com/KevinJump/uSync/issues/511 which sounds pretty similar, think we prboably need to put a bit more effort into staying upto date with patches.

Sorry I should have mentioned version in the discussion (and searched the issues / release notes better)

**Comment by KevinJump** [comment]
this is from last time we looked at it

https://github.com/KevinJump/uSync/blob/v13/main/uSync.BackOffice/SyncHandlers/Handlers/DataTypeHandler.cs#L62-L93

So the deletes of datatypes happen in post (which is after everything has synced). 

I am not sure but just changing a property from one editor to another of the same type "should't" trigger any umbraco stuff, but i wouldn't be supprised if it did.

**Comment by KevinJump** [comment]
ohh just found another bit. (just putting it here, so i have all the references for when i do look at).

https://github.com/KevinJump/uSync/blob/v13/main/uSync.Core/Serialization/Serializers/DataTypeSerializer.cs#L55-L90
